similar/misc/physfsx.cpp

```diff
@@ -226,20 +226,18 @@ bool PHYSFSX_init(int argc, char *argv[])
 	// For Macintosh, add the 'Resources' directory in the .app bundle to the searchpaths
 #if defined(__APPLE__) && defined(__MACH__)
 	{
-		ProcessSerialNumber psn = { 0, kCurrentProcess };
-		FSRef fsref;
-		OSStatus err;
-		
-		err = GetProcessBundleLocation(&psn, &fsref);
-		if (err == noErr)
-			err = FSRefMakePath(&fsref, reinterpret_cast<uint8_t *>(fullPath), PATH_MAX);
-		
-		if (err == noErr)
-		{
-			strncat(fullPath, "/Contents/Resources/", PATH_MAX + 4 - strlen(fullPath));
-			fullPath[PATH_MAX + 4] = '\0';
-			con_printf(CON_DEBUG, "PHYSFS: append resources directory \"%s\" to search path", fullPath);
-			PHYSFS_addToSearchPath(fullPath, 1);
+		CFBundleRef mainBundle = CFBundleGetMainBundle();
+		if (mainBundle) {
+			CFURLRef resourcesURL = CFBundleCopyResourcesDirectoryURL(mainBundle);
+			if (resourcesURL) {
+				if (CFURLGetFileSystemRepresentation(resourcesURL, TRUE, reinterpret_cast<uint8_t *>(fullPath), PATH_MAX + 5)) {
+					fullPath[PATH_MAX + 4] = '\0';
```

Comment:
`CFURLGetFileSystemRepresentation` says it will null-terminate. Would setting the last value of the string to nil still be desired?

Reply:
I hadn't read how `CFURLGetFileSystemRepresentation` would work.  The old code explicitly set a terminator, but if the new version gets a terminator from the system, that should be fine.  I don't like having "magic" indexes like the old code had.  If that can be eliminated by delegating the terminator to the system, that is fine.  If it could not be eliminated, then I would want to use a computed index.